python/vyos/ifconfig/interface.py

```diff
@@ -977,25 +977,76 @@ def del_tc_qdisc(self,dev,kind,handle):
             old_handle = rule['handle']
             old_kind = rule['kind']
             if old_dev == dev and old_handle == handle and old_kind == kind:
-                delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
-                self._cmd(delete_tc_cmd)
+                if 'root' in rule and rule['root']:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} root {kind}'
+                    self._cmd(delete_tc_cmd)
+                else:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
+                    self._cmd(delete_tc_cmd)
 
 
 
     def apply_mirror(self,config):
```

Comment:
There is no need to pass `config` as you always can make use of `self._config` which holds the exact same data.